rules/__tests__/valid-expect.test.js

```diff
@@ -3,16 +3,24 @@
 const RuleTester = require('eslint').RuleTester;
 const rules = require('../..').rules;
 
-const ruleTester = new RuleTester();
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 8,
+  },
+});
 
 ruleTester.run('valid-expect', rules['valid-expect'], {
   valid: [
     'expect("something").toEqual("else");',
     'expect(true).toBeDefined();',
     'expect([1, 2, 3]).toEqual([1, 2, 3]);',
     'expect(undefined).not.toBeDefined();',
-    'expect(Promise.resolve(2)).resolves.toBeDefined();',
-    'expect(Promise.reject(2)).rejects.toBeDefined();',
+    'test("foo", () => { return expect(Promise.resolve(2)).resolves.toBeDefined(); });',
+    'test("foo", async () => { await expect(Promise.reject(2)).rejects.toBeDefined(); });',
+    'test("foo", () => expect(Promise.resolve(2)).resolves.toBeDefined());',
+    'test("foo", async () => await expect(Promise.reject(2)).rejects.toBeDefined());',
+    'test("foo", async () => { expect(await Promise.resolve(2)).resolves.toBeDefined(); });',
+    'test("foo", async () => { expect(await Promise.reject(2)).rejects.toBeDefined(); });',
```

Comment:
Based on the comments in https://github.com/jest-community/eslint-plugin-jest/issues/54#issuecomment-365531253, I might remove the logic surrounding these test cases, as it's not valid to await the Promise inside `expect()` and also use `resolves` or `rejects`.

Reply:
It'll _work_, it just doesn't make any sense. So I think we should warn in that case as you're doing currently

Reply:
Actually, I was wrong, jest'll throw.

![image](https://user-images.githubusercontent.com/1404810/36297138-50978a9a-12f1-11e8-9392-922930c562f9.png)

Reply:
so if anything it should be added to `invalid` 🙂